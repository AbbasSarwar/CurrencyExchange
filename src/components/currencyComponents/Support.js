import { useDispatch, useSelector } from 'react-redux';
import { useEffect } from 'react';
import { v4 as uuidv4 } from 'uuid';
import { FetchData } from '../../redux/APIs';

const Support = () => {
  const { exchange, error, status } = useSelector((state) => state.exchange);
  const Dispatch = useDispatch();
  useEffect(() => {
    Dispatch(FetchData());
  }, [Dispatch]);
  const Ref = Object.values(exchange);
  if (error) {
    return error;
  }
  if (status) {
    return <p className="Loading">...Loading...</p>;
  }
  return (
    <section className="support">
      <div className="supportContainer">
        <h1>Support Currencies</h1>
        <ul className="supportList">
          {
            Ref.map((item) => {
              const {
                currencyName, countryCode, icon,
                availableFrom, availableUntil,
                status,
              } = item;
              return (
                <li className="supprtedshow" key={uuidv4()}>
                  <img src={icon} alt="" />
                  <p>{currencyName}</p>
                  <p>{countryCode }</p>
                  <p className="date">
                    {availableFrom}
                    {'  '}
                    -
                    {'  '}
                    {availableUntil}
                  </p>
                  <p className="status">{status}</p>
                </li>
              );
            })
          }
        </ul>
      </div>
    </section>
  );
};
export default Support;
