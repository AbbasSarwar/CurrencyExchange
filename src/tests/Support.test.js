import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import Support from './Support';

describe('Support Displat', () => {
  const exchange = {
    AMD: {
      availableFrom: '2009-09-28',
      availableUntil: '2023-05-24',
      countryCode: 'AM',
      countryName: 'Armenia',
      currencyCode: 'AMD',
      currencyName: 'Armenian Dram',
      icon: 'https://currencyfreaks.com/photos/flags/amd.png',
      status: 'AVAILABLE',
    },
    ACH: {
      availableFrom: '2009-09-28',
      availableUntil: '2023-05-24',
      countryCode: 'ACH',
      countryName: 'Global',
      currencyCode: 'CYRPTO',
      currencyName: 'Alchemy Pay',
      icon: 'https://currencyfreaks.com/photos/flags/amd.png',
      status: 'AVAILABLE',
    },
  };

  it('renders correctly', () => {
    const tree = renderer.create(
      <BrowserRouter>
        <Support exchange={exchange} error="" status={exchange.AMD.status} />
      </BrowserRouter>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
