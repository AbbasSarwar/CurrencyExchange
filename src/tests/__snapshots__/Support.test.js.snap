// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Support Displat renders correctly 1`] = `
<section
  className="support"
>
  <div
    className="supportContainer"
  >
    <h1>
      Support Currencies
    </h1>
    <ul
      className="supportList"
    >
      <li
        className="supportedshow"
      >
        <img
          alt=""
          src="https://currencyfreaks.com/photos/flags/amd.png"
        />
        <p>
          Armenian Dram
        </p>
        <p>
          AM
        </p>
        <p
          className="date"
        >
          2009-09-28
           
          -
          2023-05-24
        </p>
        <p
          className="status"
        >
          AVAILABLE
        </p>
      </li>
      <li
        className="supportedshow"
      >
        <img
          alt=""
          src="https://currencyfreaks.com/photos/flags/amd.png"
        />
        <p>
          Alchemy Pay
        </p>
        <p>
          ACH
        </p>
        <p
          className="date"
        >
          2009-09-28
           
          -
          2023-05-24
        </p>
        <p
          className="status"
        >
          AVAILABLE
        </p>
      </li>
    </ul>
  </div>
</section>
`;
